go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -1178,6 +1187,6 @@ func (bnc *BaseNetworkController) wasPodReleasedBeforeStartup(uid, nad string) b
 	return bnc.releasedPodsBeforeStartup[nad].Has(uid)
 }
 
-func isAllowedForMigration(isSecondary, isPrimaryNetwork, isl2Topology bool) bool {
-	return isSecondary && !isPrimaryNetwork && isl2Topology
+func (bnc *BaseNetworkController) isAllowedForMigration() bool {
```

Comment:
I'd have this as a function instead. There's no need for the layer3 controller to have this method for instance.

Also, should't localnet be also a "customer" of this feature ?... The exact same mechanism should be re-used there to improve the migration downtime, right ?

Reply:
Having a function that is going to receive NetInfo is kind of the same also if will be in the same context as layer3, also I am total agains a free function receiving random booleans.

Reply:
I don't understand what you mean, but FWIW, I also don't want a function that receives random booleans.